specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2623 @@
+          }
+        },
+        "progress" : {
+          "description" : "Progress information for the namespace hydration",
+          "type" : "object",
+          "properties" : {
+            "blobCount" : {
+              "description" : "The number of Blobs processed.",
+              "type": "integer"
+            },
+            "size" : {
+              "type" : "integer",
+              "description" : "Aggregate size of blobs ingested so far"
+            },
+            "rate" : {
+              "type" : "integer",
+              "description": "Rate of ingest accounting for the last 30 seconds"
+            },
+            "blobName" : {
+              "type" : "string",
+              "description" : "Name of the current blob"
+            },
+            "errorCount" : {
+              "type" : "integer",
+              "description" :"Number of failures"
+            }
+          }
+        }
+      }
+    },
+    "amlFilesystemArchiveInfo": {
+      "type":"object",
+      "description": "Information required to execute the archive operation",
+      "properties": {
+        "filesystemPath" : {
+          "type" : "string",
+          "description" : "The path to the filesystem to archive."
+        }
+      }
+    },
+    "amlFilesystemArchive" : {
+      "description" : "Information about the amlFilesystem archive",
+      "type" : "object",
+      "readOnly": true,
+      "properties": {
+        "filesystemPath" : {
+          "readOnly": true,
+          "type" : "string",
+          "description" : "The path to the filesystem to archive."
+        },
+        "status" : {
+          "description" : "The status of the archive",
+          "type" : "object",
+          "readOnly" : true,
+          "properties": {
+            "state" : {
+              "readOnly": true,
+              "description" : "The state of the archive operation",
+              "type" : "string",
+              "enum" : [
+                "Idle",
+                "InProgress"
+              ],
+              "x-ms-enum": {
+                "name": "ArchiveStatusType",
+                "modelAsString": true
+              }
+            },
+            "lastCompletionTime" : {
+              "readOnly" : true,
+              "description" : "The time of the last completed archive operation",
+              "type" : "string",
+              "format" : "date-time"
+            },
+            "lastStartedTime" : {
+              "readOnly" : true,
+              "description" : "The time the latest archive operation started",
+              "type" : "string",
+              "format" : "date-time"
+            },
+            "percentComplete" : {
+              "readOnly" : true,
+              "description" : "The completion percentage of the archive",
+              "type" : "integer"
+            },
+            "errorCode" : {
+              "readOnly" : true,
+              "description" : "The error code of the archive operation",
```

Comment:
What is the space of this? IOW, when I read this in the docs, how do I know how to translate the code to something meaningful? Is this an HTTP status code? Linux errno? POSIX errno?

Reply:
The codes would be used to quickly check for specific errors in client code that you would program to without needing to do string comparisons.  Message string that are composed at run time would be ever more difficult to code against.  The message is nice for logging or interactive work, but programmatic failures are handled easier by some code.

In the REST Api guidelines error objects returned by operations must return code and message in the error object.  In that case code is a string.
https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses

Since this is a status in the resource for the archive action, its not exactly that but I can modify it.  I could use code and message and make code a string.  Is there going to be a way to save both the message and code from a failed archive?
Let me know what we'll have an I can modify it to match. 


Specifics from that link on code/message to save time.
The value for the "code" name/value pair is a language-independent string. Its value is a service-defined error code that SHOULD be human-readable. This code serves as a more specific indicator of the error than the HTTP error code specified in the response. Services SHOULD have a relatively small number (about 20) of possible values for "code," and all clients MUST be capable of handling all of them. Most services will require a much larger number of more specific error codes, which are not interesting to all clients. These error codes SHOULD be exposed in the "innererror" name/value pair as described below. Introducing a new value for "code" that is visible to existing clients is a breaking change and requires a version increase. Services can avoid breaking changes by adding new error codes to "innererror" instead.

The value for the "message" name/value pair MUST be a human-readable representation of the error. It is intended as an aid to developers and is not suitable for exposure to end users. Services wanting to expose a suitable message for end users MUST do so through an annotation or custom property. Services SHOULD NOT localize "message" for the end user, because doing so might make the value unreadable to the app developer who may be logging the value, as well as make the value less searchable on the Internet.

Reply:
I'm not asking for a change to what the RP generates here - I'm asking for the description to give the developer enough info to find the relevant documentation. For example, just saying "HTTP status" is good enough. If it's some RP-specific thing, I suggest a URL to the relevant document.

Reply:
I was thinking this was from lustre's archive operation more than the RP.  Can the archive fail for any reason?
We have states NotConfigured, Idle, and InProgress.  Cancelling is being looked into.  We don't have any failure state.  If there isn't going to be a failed state, this field may not be needed along with the errorMessage field.

Reply:
These are visible to the customer, so they must make sense to the customer. This will also help us with testing.

Reply:
Switched to string type and used the Server-defined description you provided.  I added Server-defined to the errorMessage as well.